traveler-ios-ui/TravelerKitUI/ViewControllers/ReceiptViewController.swift

```diff
@@ -42,6 +42,9 @@ class ReceiptViewController: UITableViewController {
         let cell = tableView.dequeueReusableCell(withIdentifier: infoCellIdentifier, for: indexPath) as! InfoCell
         let product = receipt!.order.products[indexPath.row]
         cell.titleLabel.text = product.title
+        if let bookable = product as? BookableProduct {
+            cell.valueLabel.text = DateFormatter.dateOnlyFormatter.string(from: bookable.eventDate)
+        }
```

Comment:
Its best if this view controller doesnt know of BookableProduct.

you can instead define an extension (in a separate file I would say for this)

```
extension Product {
  var detailDescription: String? {
      switch self {
      case is BookableProduct:
         ....
      }
  }
}
```

Reply:
Or maybe even add it to the core if you think it makes sense and its not even an extension anymore and part of the protocol